src/components/PeopleTable.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useMemo } from "react"
+import { Person } from "../types"
+import { PersonLink } from "./PersonLink"
+import classNames from "classnames"
+
+type Props = {
+  people: Person[];
+  selectedSlug: string | undefined;
+}
+
+const tableHeaders = ['Name', 'Sex', 'Born', 'Died', 'Mother', 'Father'];
+
+export const PeopleTable: React.FC<Props> = ({ people, selectedSlug }) => {
+  if (people.length === 0) {
+    return (
+      <p data-cy="noPeopleMessage">There are no people on the server</p>
+    );
+  }
+
+  const preparedPeople = useMemo(() => {
+    return people.map(person => {
+      const getParentElement = (parentName: string | null) => {
+        if (!parentName) {
+          return '-'
+        }
+        const parent = people.find(per => per.name === parentName)
+        return parent ? <PersonLink person={parent} /> : parentName
+      }
+      return {
+        ...person,
+        motherElement: getParentElement(person.motherName),
+        fatherElement: getParentElement(person.fatherName),
+      };
+    });
```

Comment:
- Create a function and move to utils/helpers.
- Don't store react components in the object. Just mother/father objects instead